plugins/catalog-backend-module-incremental-ingestion/src/router/routes.ts

```diff
@@ -15,218 +15,269 @@
+          });
+        } else {
+          res.status(404).json({
+            success: false,
+            message: `Provider '${provider}' not found`,
+          });
+        }
       }
-    }
-  });
-
-  // Stop the provider and pause it for 24 hours
-  router.post(`${PROVIDER_BASE_PATH}/cancel`, async (req, res) => {
-    const { provider } = req.params;
-    const record = await manager.getCurrentIngestionRecord(provider);
-    if (record) {
-      const next_action_at = new Date();
-      next_action_at.setTime(next_action_at.getTime() + 24 * 60 * 60 * 1000);
-      await manager.updateByName(provider, {
-        next_action: 'nothing (done)',
-        ingestion_completed_at: new Date(),
-        next_action_at,
-        status: 'resting',
-      });
+    });
+
+    // Wipe out all ingestion records for the provider and pause for 24 hours
+    router.delete(PROVIDER_BASE_PATH, async (req, res) => {
+      const { provider } = req.params;
+      const result = await this.manager.purgeAndResetProvider(provider);
+      res.json(result);
+    });
+
+    // Get the ingestion marks for the current cycle
+    router.get(`${PROVIDER_BASE_PATH}/marks`, async (req, res) => {
+      const { provider } = req.params;
+      const record = await this.manager.getCurrentIngestionRecord(provider);
+      if (record) {
+        const id = record.id;
+        const records = await this.manager.getAllMarks(id);
+        res.json({ success: true, records });
+      } else {
+        const providers: string[] = await this.manager.listProviders();
+        if (providers.includes(provider)) {
+          this.logger.debug(`${provider} - Ingestion record found`);
+          res.json({
+            success: true,
+            message: 'No records yet (provider is restarting)',
+          });
+        } else {
+          this.logger.error(
+            `${provider} - No ingestion record found in the database!`,
+          );
+          res.status(404).json({
+            success: false,
+            status: {},
+            last_error: `Provider '${provider}' not found`,
+          });
+        }
+      }
+    });
+
+    router.delete(`${PROVIDER_BASE_PATH}/marks`, async (req, res) => {
+      const { provider } = req.params;
+      const deletions = await this.manager.clearFinishedIngestions(provider);
+
       res.json({
         success: true,
-        message: `${provider}: Current ingestion canceled.`,
+        message: `Expired marks for provider '${provider}' removed.`,
+        deletions,
       });
-    } else {
-      const providers: string[] = await manager.listProviders();
-      if (providers.includes(provider)) {
-        logger.debug(`${provider} - Ingestion record found`);
-        res.json({
-          success: true,
-          message: 'Provider is currently restarting, please wait.',
-        });
-      } else {
-        res.status(404).json({
+    });
+
+    router.post(`${PROVIDER_BASE_PATH}/delta`, async (req, res) => {
```

Comment:
Hm, is this for debugging? How is this one used?

Reply:
I'm going to rename `delta` to `event`, FYI.

This route is the mechanism for picking up events targeting the entity provider. It's designed so that every incremental entity provider is already wired up for receiving events (such as those from subscribed webhooks), and only needs to handle the event passed to it in the `onEvent` method.

Reply:
This route is gone, now that I'm relying on the events backend.